### DISCUSSION TITLE
Keycloack pod gives internal server error(K8s)

### DISCUSSION THREAD

**Discussion by AdnanKhan01** [keycloak/keycloak#29283] [Q&A]
Hello everyone,
I am using keycloak on k8s.  I encountered a situation in which the pod was running perfectly fine but when the user would login through the app , it would give internal server error. I checked the logs and everything but couldn't find the cause. But when I deleted the pod and the postreSQL pod on which the keycloak pod was dependent upon and then recreated them everything started working fine.
My questions please
1. Was it because the DB was stuck?
2. What actually caused the internal server error and when recreated everything worked fine?
3. Is there any way i can check logs of that incident because the pod itself was deleted and new pod was created
Thanks guys

**Comment by shawkins** [comment]
> Was it because the DB was stuck?

Unfortunately it's impossible to say with detailed logging.

> What actually caused the internal server error and when recreated everything worked fine?

If the internal server error doesn't have sufficient logging, then try increasing the log level on the pod via the log-level option - unfortunrately a restart is required

> Is there any way i can check logs of that incident because the pod itself was deleted and new pod was created

Kubernetes will keep the previos pod logs, see "kubectl logs <podname> --previous". However anything earlier than that and you'll need to use a logging stack that save things for longer.
